### ISSUE TITLE
qemu hangs on black screen

### ISSUE THREAD

**Issue by marprok** [SerenityOS/serenity#4972]
I am trying to setup serenity again after a long time on my  Ubuntu(20.04.1 LTS (Focal Fossa) 64-bit) installation and since cmake failed due to https://github.com/SerenityOS/serenity/issues/4538, I tried ninja for the first time in my life.

First of all I have all the listed dependencies installed as you can see bellow,
![image](https://user-images.githubusercontent.com/18293204/104808953-3e11fb00-57f2-11eb-8590-35059eab83b9.png)

Then I build the tool-chain with no visible errors and tried to run `ninja` -> `ninja install` -> `ninja image` -> `ninja run` following the build instructions step by step inside an empty Build directory.

I did not encounter any errors during any of the above commands and I assumed that everything was fine. After `ninja run`, qemu just hangs on a black screen and bellow you can find the corresponding logs,
```
marios@void:~/Source/serenity/Build$ ninja run
[0/2] Re-checking globbed directories...
[0/1] cd /home/marios/Source/serenity/...ome/marios/Source/serenity/Meta/run.sh
qemu-system-i386: multiboot knows VBE. we don't
[Kernel]: CPU[0]: Supported features: nx pae pge sse tsc rdtscp sep syscall mmx sse2 sse3 ssse3 sse4.1 sse4.2
[Kernel]: CPU[0]: No RDRAND support detected, randomness will be poor
[Kernel]: CPU[0]: Physical address bit width: 36
[Kernel]: Installing Unhandled Handlers
[Kernel]: MM: boot_pdpt @ P0x00109000
[Kernel]: MM: boot_pd0 @ P0x0010a000
[Kernel]: MM: boot_pd3 @ P0x0010b000
[Kernel]: MM: kernel range: P0x00100000 - P0x00281000
[Kernel]: MM: kmalloc range: P0x00281000 - P0x00682000
[Kernel]: MM: Multiboot mmap: base_addr = 0x00000000, length = 0x0009fc00, type = 0x1
[Kernel]: MM: Multiboot mmap: base_addr = 0x0009fc00, length = 0x00000400, type = 0x2
[Kernel]: MM: Multiboot mmap: base_addr = 0x000f0000, length = 0x00010000, type = 0x2
[Kernel]: MM: Multiboot mmap: base_addr = 0x00100000, length = 0x1fee0000, type = 0x1
[Kernel]: MM: Multiboot mmap: base_addr = 0x1ffe0000, length = 0x00020000, type = 0x2
[Kernel]: MM: Multiboot mmap: base_addr = 0xfeffc000, length = 0x00004000, type = 0x2
[Kernel]: MM: Multiboot mmap: base_addr = 0xfffc0000, length = 0x00040000, type = 0x2
[Kernel]: Super physical region: P0x00700000 - P0x007ff000
[Kernel]: User physical region: P0x00800000 - P0x1ffe0000
[Kernel]: Early access to ACPI tables for interrupt setup
[Kernel]: Interrupts: Switch to Legacy PIC mode
[Kernel]: PIC(i8259): cascading mode, vectors 0x50-0x5f
[Kernel]: Interrupts: Detected Dual i8259
[Kernel]: ACPI: Using RSDP @ P0x000f5a60
[Kernel]: ACPI: Main Description Table valid? true
[Kernel]: ACPI: Using RSDT, Enumerating tables @ P0x1ffe15c9
[Kernel]: ACPI: RSDT Revision 1, Total length - 48
[Kernel]: ACPI: Initializing Fixed ACPI data
[Kernel]: ACPI: Searching for the Fixed ACPI Data Table
[Kernel]: ACPI: Fixed ACPI data, Revision 1, Length 116 bytes
[Kernel]: ACPI: DSDT P0x1ffe0040
[Kernel]: ACPI: Dynamic Parsing Enabled, Can parse AML
[Kernel]: I8042: Dual channel controller
[Kernel]: PS2MouseDevice: Mouse wheel enabled!
[Kernel]: PS2MouseDevice: 5 buttons enabled!
[Kernel]: Starting SerenityOS...
[Kernel]: RTC: Year: 2021, month: 1, day: 16, hour: 10, minute: 0, second: 22
[Kernel]: HPET @ P0x1ffe1591
[Kernel]: HPET: Minimum clock tick - 0
[Kernel]: HPET: Timers count - 3
[Kernel]: HPET: Main counter size: 64 bit
[Kernel]: HPET: Timer[0] comparator size: 64 bit mode: 64 bit
[Kernel]: HPET: Timer[1] comparator size: 64 bit mode: 64 bit
[Kernel]: HPET: Timer[2] comparator size: 64 bit mode: 64 bit
[Kernel]: HPET: frequency 100000000 Hz (100 MHz) resolution: 10ns
[Kernel]: HPET: Setting appropriate functions to timers.
[Kernel]: Time: Scanning for periodic timers
[Kernel]: Time: Scanning for non-periodic timers
[Kernel]: VC: Switch to 0 (0xc03c783c)
[colonel(0:0)]: Scheduler[0]: idle loop running
[init_stage2(1:1)]: Testing PCI via manual probing... 
[init_stage2(1:1)]: PCI IO Supported!
[init_stage2(1:1)]: PCI: Using I/O instructions for PCI configuration space access
[init_stage2(1:1)]: PCI [0000:00:00:00] (8086:1237)
[init_stage2(1:1)]: PCI [0000:00:01:00] (8086:7000)
[SyncTask(2:2)]: SyncTask is running
[init_stage2(1:1)]: PCI [0000:00:01:01] (8086:7010)
[init_stage2(1:1)]: PCI [0000:00:01:02] (8086:7020)
[init_stage2(1:1)]: PCI [0000:00:01:03] (8086:7113)
[init_stage2(1:1)]: PCI [0000:00:02:00] (1234:1111)
[init_stage2(1:1)]: PCI [0000:00:03:00] (8086:2922)
[init_stage2(1:1)]: PCI [0000:00:04:00] (8086:100e)
[init_stage2(1:1)]: BXVGA: framebuffer @ P0xf8000000
[init_stage2(1:1)]: BXVGADevice resolution set to 1024x768 (pitch=4096)
[init_stage2(1:1)]: UHCI: Controller found (8086:7020) @ PCI [0000:00:01:02]
[init_stage2(1:1)]: UHCI: I/O base IO c040
[init_stage2(1:1)]: UHCI: Interrupt line: 11
[init_stage2(1:1)]: UHCI: Allocated framelist at physical address P0x00700000
[init_stage2(1:1)]: UHCI: Framelist is at virtual address V0xc094c000
Done!
UHCI: QH(0xc094e000) @ 0x00701000: link_ptr=0x00701022, element_link_ptr=0x00000001
UHCI: QH(0xc094e020) @ 0x00701020: link_ptr=0x00701042, element_link_ptr=0x00000001
UHCI: QH(0xc094e040) @ 0x00701040: link_ptr=0x00701062, element_link_ptr=0x00000001
UHCI: QH(0xc094e060) @ 0x00701060: link_ptr=0x00701082, element_link_ptr=0x00000001
UHCI: QH(0xc094e080) @ 0x00701080: link_ptr=0x00704001, element_link_ptr=0x00000001
[init_stage2(1:1)]: UHCI: Reset completed!
[init_stage2(1:1)]: UHCI: Started!
[init_stage2(1:1)]: E1000: Found @ PCI [0000:00:04:00]
[init_stage2(1:1)]: E1000: port base: IO c000
[init_stage2(1:1)]: E1000: MMIO base: P0xfeb80000
[init_stage2(1:1)]: E1000: MMIO base size: 131072 bytes
[init_stage2(1:1)]: E1000: Interrupt line: 11
[init_stage2(1:1)]: E1000: Has EEPROM? true
[init_stage2(1:1)]: E1000: MAC address: 52:54:00:12:34:56
[init_stage2(1:1)]: Syscall: int 0x82 handler installed
[init_stage2(1:1)]: SB16: found version 4.5
[init_stage2(1:1)]: SB16: IRQ 5
[init_stage2(1:1)]: VMWareBackdoor: Enabling absolute mouse mode
[init_stage2(1:1)]: IDEChannel: Bus master IDE: IO c080
[init_stage2(1:1)]: IDEChannel: Name=QEMU HARDDISK, C/H/Spt=180/16/63
[init_stage2(1:1)]: IDEChannel: Bus master IDE: IO c088
[init_stage2(1:1)]: Master Boot Record: mbr_signature=0x0
[init_stage2(1:1)]: Master Boot Record: invalid signature
[init_stage2(1:1)]: VFS: mounted root from Ext2FS (device:3,0 (PATADiskDevice))
[init_stage2(1:1)]: Loading kernel symbol table...
```
And for `ninja install`,
```
marios@void:~/Source/serenity/Build$ ninja install
[0/2] Re-checking globbed directories...
[0/1] Install the project...
-- Install configuration: ""
```

The last time I built serenity on this machine was when I made https://github.com/SerenityOS/serenity/pull/4011 and that time I built it using cmake with no problems.

What am I doing wrong here? I know that since the automatic builds succeed all the time it must be something that I am doing wrong with ninja.

**Comment by bcoles** [comment]
This looks like #4670.

Do you have KVM enabled? (does `/dev/kvm` exist?)

**Comment by tomuta** [comment]
I can boot just fine with both KVM enabled and disabled. But maybe it depends on how fast your machine is, maybe it's a weird timing problem.

**Comment by marprok** [comment]
> This looks like #4670.
> 
> Do you have KVM enabled? (does `/dev/kvm` exist?)
> 
> I have Serenity latest `master` running ok in Qemu on Ubuntu 20.04 with KVM.

I do have kvm enabled,
```
marios@void:~$ sudo kvm-ok
[sudo] password for marios: 
INFO: /dev/kvm exists
KVM acceleration can be used
```

But if it is a weird timing issue(this machine is old by today's standards) maybe with a little guidance I can debug it since I can reproduce it @tomuta 

Below you can find the specs for this machine,
![image](https://user-images.githubusercontent.com/18293204/104822790-e3a68800-584d-11eb-8ce2-4060edaf7e36.png)

**Comment by marprok** [comment]
After some debug logs, I found that the actual hanging happens in Process.cpp::Process::create_user_process at the 
`error = process->exec(path, move(arguments), move(environment));`. For some reason which I haven't found out yet, the exec for `/bin/SystemServer` hangs the system. I will continue the investigation in my spare time but it seems strange that this happens only to me...

**Comment by ADKaster** [comment]
I've seen this before but generally if you click on the qemu window and wave the mouse or smash the keyboard a bit it should find some good entropy. 

See https://github.com/SerenityOS/serenity/issues/4490 

Might have to do with which qemu settings are being used/available?

**Comment by marprok** [comment]
> I've seen this before but generally if you click on the qemu window and wave the mouse or smash the keyboard a bit it should find some good entropy.
> 
> See #4490
> 
> Might have to do with which qemu settings are being used/available?

Thank you very much for the reply! I did as you instructed and I got the cute serenity terminal showing up again :P . In my case though, key pressing didn't seem to do the trick but instead when I move the mouse for ~4 seconds it seems to build up enough entropy and boot successfully.

Maybe we should change the issue label to `bug` instead of `build issue` since it was not one.
Also, should I close the issue since there is another issue related to it?
